### ISSUE TITLE
Add support for structural types

### ISSUE THREAD

**Issue by densh** [scala-native/scala-native#312]
Calling methods that are defined in structural types isn't supported yet:

``` Scala
val x: { def foo(): Unit } = new { def foo() = println("hi") }
x.foo()
```

**Comment by densh** [comment]
As discussed yesterday, here is the checklist of the things to do here:

- [ ] Add support for dynamic methods to NIR
    - [ ] Add new `DynMethod` op
    - [ ] Update BinarySerializer
    - [ ] Update BinaryDeserializer
    - [ ] Update Tags
    - [ ] Update Shows
- [ ] Implement [front-end stub](https://github.com/scala-native/scala-native/blob/master/nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala#L847) to emit new opcodes 
- [ ] Implement a new `DynamicMethodLowering` pass
- [ ] Update [`RuntimeTypeInfoInjection`](https://github.com/scala-native/scala-native/blob/master/tools/src/main/scala/scala/scalanative/compiler/pass/RuntimeTypeInfoInjection.scala) to emit additional metadata

**Comment by densh** [comment]
Fixed by @LukasKellenberger in #460
